pyroSAR/snap/S1_SLC/S1_HA_proc_FUN.py

```diff
@@ -0,0 +1,416 @@
+        extDEM: bool
+            set to true if external DEM should be used in processing
+        ext_DEM_noDatVal: int or float
+            dependent on external DEM, default False
+        ext_DEM_file: str
+            path to file of external DEM, must be a format that SNAP can handle
+        msk_NoDatVal: bool
+            if true No data values of DEM, especially at sea, are masked out
+        ext_DEM_EGM: bool
+            apply earth gravitational model to external DEM, default true
+        imgResamp: str
+            image resampling method, must be supported by SNAP
+        demResamp: str
+            DEM resampling method, must be supported by SNAP
+        speckFilter: str
+            type of speckle filtering approach, default is Box Car Filter
+        ml_RgLook: int
+            number of looks in range, default is 4
+        ml_AzLook: int
+            number of looks in azimuth, default is 1
+        firstBurstIndex: int or None
+            index of first burst for TOPSAR-split
+        lastBurstIndex: int or None
+            index of last burst for TOPSAR-split
+        clean_tmpdir, bool
+            delete tmpdir, default true
+        osvPath: None
+            specify path to locally stored OSVs, if none default OSV path of SNAP is set
+
+        Returns
+        -------
+        Raster files of selected output format for selected H-alpha features
+
+        Note
+        ----
+        Only set first and last burstindex if all files you are processing have the same number of bursts
+
+        Examples
+        --------
+        process all H-alpha features for given SLC file
+
+        >>> from pyroSAR.snap import S1_HA_proc
+        >>> filename= 'S1A_IW_GRDH_1SDV_20180829T170656_20180829T170721_023464_028DE0_F7BD.zip'
+        >>> gpt_paras = ["-e", "-x", "-c","35G", "-q", "16", "-J-Xms25G", "-J-Xmx75G"]
+        >>> decompFeats= ["Alpha", "Entropy", "Anisotropy"]
+        >>> S1_HA_proc(infiles= filename, gtp_paras= gpt_paras, decompFeats= decompFeats)
+
+    """
+    
+    ##define formatName for reading zip-files
+    formatName = "SENTINEL-1"
+    ##specify tmp output format
+    tpm_format = "BEAM-DIMAP"
+    ## create temp dir for intermediate .dim files
+    if tmpdir is None:
+        tmpdir = os.path.join(os.getcwd(), "tmp_dim")
+        if os.path.isdir(tmpdir) == False:
+            os.mkdir(tmpdir)
+    ##check if a single IW or consecutive IWs are selected
+    if isinstance(IWs, str):
+        IWs = [IWs]
+    if sorted(IWs) == ["IW1", "IW3"]:
+        raise RuntimeError("Please select single or consecutive IW")
+    
+    ##extract info about files and order them by date
+    ##handle length and type of infiles: str or list
+    if isinstance(infiles, str):
+        info = pyroSAR.identify(infiles)
+        fps_lst = [info.scene]
+        info = [info]
+    elif isinstance(infiles, list):
+        info = pyroSAR.identify_many(infiles, verbose=False, sortkey='start')
```

Comment:
I am not a SAR expert, but at least I can point few bugs on top of pySAR v0.18.0

verbose variable does not exists on pyroSAR.identify_many